docs/concepts/results.md

```diff
@@ -1,7 +1,89 @@
 # TimeEval results
 
-How to work with the results of TimeEval.
+Upon executing TimeEval python script for the given datasets, timeeval checks the effectiveness of selected algorithm and generate an output file as `./TimeEval/results/<timestamp>` directory. The output directory has following data structure:
+
+```bash
+results/<time-stamp>/
+├── results.csv
+├── algorithm_1/hyperparameter/
+|   ├── dataset_1/epoch/
+│   |   ├── anomaly_scores.ts
+│   |   ├── docker-algorithm-scores.csv
+│   |   ├── execution.log
+│   |   ├── hyper_params.json
+│   |   └── metrics.csv
+|   └── dataset_2/epoch/
+│       ├── anomaly_scores.ts
+│       ├── docker-algorithm-scores.csv
+│       ├── execution.log
+|	├── model.pkl
+│       ├── hyper_params.json
+│       └── metrics.csv
+└── algorithm_2/hyperparameter/
+    └── dataset_1/epoch/
+        ├── anomaly_scores.ts
+        ├── docker-algorithm-scores.csv
+        ├── execution.log
+        ├── hyper_params.json
+        └── metrics.csv
 
-```{warning}
-WIP
 ```
+Description of each file is given below. 
+
+## result.csv
+The file `result.csv` contains following attributes:
+
+| Column Name | Datatype | Description |
+| --- | --- | --- |
+| algorithm| str | applied on the dataset |
+| collection| str | indicates the name of the dataset | 
+| dataset| str | indicates the targeted segment of the data collection |
+| algo_training_type | str | three learning types: unsupervised, semi-supervised, supervised |
+| algo_input_dimensionality | str | two kinds: univarites, multivariate |
+| dataset_training_type | str | three learning type as above | 
+| dataset_input_dimensionality | str | two kinds as above |
+| train_preprocess_time| float64| |
+| train_main_time| float64 | |
+| execute_preprocess_time| float64 | |
+| execute_main_time | float64 | time taken for execute the script for the given datafile |
+| execute_postprocess_time|  float64 | |
+| status| str | assigns flag value upon the process |
+| error_message| str | |
+| repetition| int | number of times dataset being processed |
+| hyper_params| float64 | for given algorithm |
+| hyper_params_id| float64 | 32 alphanumerical characters |
+| metric_1| float64 | --user-- |
+| metric_2|  float64 | --user-- |
```

Comment:
- algorithm: name of the algorithm as defined in :class:`~timeeval.Algorithm`'s `name` attribute
- collection: name of the dataset collection. A collection groups similar datasets.
- dataset: name of the dataset
- dataset_training_type: specifies, whether a dataset has a training time series with anomaly labels (supervised), with normal data only (semi-supervised), or no training time series at all (unsupervised)
- dataset_input_dimensionality: specifies if the dataset has multiple channels (multivariate) or not (univariate)
- algo_training_type: specifies, whether an algorithm requires training data with anomalies (supervised), without normal data only (semi-supervised), or does not require training data (unsupervised)
- algo_input_dimensionality: univariate or multivariate (see above)
- train_preprocess_time: runtime of the preprocessing step during training in seconds
- train_main_time: runtime of the training in seconds (does not include pre-processing time)
- execute_preprocess_time: runtime of the preprocessing step during execution in seconds
- execute_main_time: runtime of the execution of the algorithm on the test time series in seconds (does not include pre- or post-processing times)
- execute_postprocess_time: runtime of the post-processing step during execution
- status: specifies, whether the algorithm executed successfully (:obj:`~timeeval.Status.OK`), exceeded the time limit (:obj:`~timeeval.Status.TIMEOUT`), exceeded the memory limit (:obj:`~timeeval.Status.OOM`), or failed (:obj:`~timeeval.Status.ERROR`)
- error_message: optional detailed error message
- repetition: repetition number if a dataset-hyperparameter-dataset combination was executed multiple times
- hyper_params: actual hyperparameter values for this execution
- hyper_params_id: alphanumerical hash of the hyperparameter configuration
- metric_1: value of the first performance metric
- ...